### ISSUE TITLE
support for oreo patch

### ISSUE THREAD

**Issue by ric96** [robherring/generic_device#7]
I've tried building after oreo, seems to be having issues with nativewindow on mesa.
Peter has update the rpi mesa for oreo, you may want to take a look. I'll issue a patch if I get the time, it takes about 5hrs for my system to finish a build.
https://github.com/android-rpi/external_mesa3d/commit/0326dac5afb98746b03576a734f1fdc8c2990d7c

**Comment by montvid** [comment]
Yes, the build for linaro_arm_userdebug is broken too with the same error. How can I fix this?

> external/mesa3d/include/EGL/eglplatform.h:100:10: fatal error: 'android/native_window.h' file not found

**Comment by montvid** [comment]
@ric96 Is copy pasting the changed files from your link enough to solve the problem?

**Comment by montvid** [comment]
Found a fix here, I used the second link:
[https://patchwork.freedesktop.org/patch/173342/](https://patchwork.freedesktop.org/patch/173342/)
[https://github.com/KhronosGroup/EGL-Registry/pull/22/files](https://github.com/KhronosGroup/EGL-Registry/pull/22/files)

**Comment by robherring** [comment]
I need to update the mesa branch. This is fixed in mesa master. Once I get things cleaned up and working again, I will push the updates.

**Comment by montvid** [comment]
thanks @robherring while compiling linaro_arm-userdebug i applied [https://github.com/KhronosGroup/EGL-Registry/pull/22/files](https://github.com/KhronosGroup/EGL-Registry/pull/22/files) patch and it compiled without errors.
